src/lib.rs

```diff
@@ -68,7 +68,7 @@
     //missing_docs,
     private_doc_tests,
     rust_2018_idioms,
-    single_use_lifetimes,
+    //single_use_lifetimes,
```

Comment:
Sigh. I had hoped this can be done without it.

Do you think `deny(single_use_lifetimes)` is useful in general? If not, we can just remove this. Otherwise, some `#[!allow...` is needed instead of just removing it globally.

Reply:
I'm afraid these `allow` would be needed on each implementation of the `RequestConnection` trait. I think we can go without `single_use_lifetimes`. I believe it won't be that bad if an avoidable single-use lifetime steps in.